src/module_2/EDA.md

```diff
@@ -0,0 +1,324 @@
+
+
+```python
+# Explorar 'order_id'
+print("Número de órdenes únicas:", len(df['order_id'].unique()))
+print(df['order_id'].value_counts())
+
+# Explorar 'user_id'
+print("Número de usuarios únicos:", len(df['user_id'].unique()))
+print(df['user_id'].value_counts())
+```
+
+Se observa que hay usuarios repetidos que realizan más de un pedido. Estas variables se toman como categóricas porque son un identificador -> ¿Luego las codifico meterlas al modelo con un label encoding por ejemplo?
+
+
+```python
+# Histograma de 'created_at'
+df['created_at'].hist(bins=20, edgecolor='black')
+plt.xlabel('Fecha y Hora de Creación')
+plt.ylabel('Frecuencia')
+plt.title('Distribución de Registros por Fecha y Hora de Creación')
+plt.show()
+
+# Histograma de 'order_date'
+df['order_date'].hist(bins=20, edgecolor='black')
+plt.xlabel('Fecha de Pedido')
+plt.ylabel('Frecuencia')
+plt.title('Distribución de Pedidos por Fecha')
+plt.show()
+```
+
+Hay una tendencia ascendente de la fecha del pedido y la fecha y hora de creación desde el 10-2020 hasta el 03-2021. Esto indica que se han comprado mayor cantidad de productos últimamente o que, la mayor parte de los datos se concentran en fechas de 2021.
+
+**2. Información del producto y variante:**
+
+- 'variant_id': Identificador único de la variante del producto. (int64)
+- 'product_type': Tipo de producto. (object)
+- 'vendor': Proveedor del producto. (object)
+- 'normalised_price': Precio normalizado del producto. (float64)
+- 'discount_pct': Porcentaje de descuento aplicado. (float64)
+
+Vemos que hay 976 variantes únicas de los productos. En este sentido, las más vistas aparecen 3446 veces mientras que la menos vista aparece únicamente 25 veces en el DataSet.
+
+
+```python
+df['variant_id'] = df['variant_id'].astype('object')
+print("Número de variantes únicas:", len(df['variant_id'].unique()))
+print(df['variant_id'].value_counts())
+```
+
+Vemos el número de tipos de productos distintos y de vendedores que existen. Existen 62 tipos de productos distintos y 264 vendedores, destacando entre los productos los alimentos enlatados como los más vistos, seguido de condimentos y aderezos y, a su vez, de arroz, pasta y legumbres.
+En cuanto a las marcas más vistas destacan biona, ecover y method.
+
+
+```python
+print("Número de tipos de producto únicos:", len(df['product_type'].unique()))
+print(df['product_type'].value_counts())
+
+print("Número de tipos de vendedores:", len(df['vendor'].unique()))
+print(df['vendor'].value_counts())
+```
+
+Al analizar el precio normalizado, su valor máximo es 1, siendo este el producto más caro. Los precios están normalizados entre 0 y 1 para facilitar las comparaciones probablemente. En cuanto al descuento, hay cifras de descuento menores a cero y mayores a 1 lo cual, a priori, no tendría sentido, estaríamos hablando de un descuento superior al 100% y no tiene sentido hablar de un descuento negativo.
```

Comment:
Why not? It depends on how it's computed. 

Let's say that RRP is 2€ and I sell it at 4€ or 1€. What would be the discount_pct on each case?